### PR TITLE
feat(label): adds brain label

### PR DESCRIPTION
**Pull request by thatsamsonkid**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our
      guidelines: https://github.com/goetzrobin/spartan/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Which package are you modifying?

- [ ] accordion
- [ ] alert
- [ ] alert-dialog
- [ ] aspect-ratio
- [ ] avatar
- [ ] badge
- [ ] button
- [ ] calendar
- [ ] card
- [ ] checkbox
- [ ] collapsible
- [ ] combobox
- [ ] command
- [ ] context-menu
- [ ] data-table
- [ ] date-picker
- [ ] dialog
- [ ] dropdown-menu
- [ ] hover-card
- [ ] input
- [x] label
- [ ] menubar
- [ ] navigation-menu
- [ ] popover
- [ ] progress
- [ ] radio-group
- [ ] scroll-area
- [ ] select
- [ ] separator
- [ ] sheet
- [ ] skeleton
- [ ] slider
- [ ] switch
- [ ] table
- [ ] tabs
- [ ] textarea
- [ ] toast
- [ ] toggle
- [ ] tooltip
- [ ] typography

## What is the current behavior?

Label component currently does not have any logic attached currently no auto generated id. But we've found there is some light logic usually required of labels for use in other components in library.

Closes #

## What is the new behavior?

Label component will auto generate a unique id unless an id is provided by user. This is especially helpful in other components where we need some sort of Id for ADA purposes for example.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: libs/ui/label/brain/src/lib/brn-label.directive.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { Directive, HostBinding, Input, signal } from '@angular/core';
+
+let nextId = 0;
+
+@Directive({
+	selector: '[brnLabel]',
+	standalone: true,
+})
+export class BrnLabelDirective {
+	@HostBinding('id')
```
Comment (elite-benni) [inline libs/ui/label/brain/src/lib/brn-label.directive.ts:10]: could this be done with a Host property?

**Comment by goetzrobin** [comment]
Can we add this fix from @elite-benni, @thatsamsonkid? And then we just need to resolve the merge conflicts and then we're good 👌🚀

**Comment by thatsamsonkid** [comment]
I'll fix merge conflict tonight!

**Comment by thatsamsonkid** [comment]
Sorry messed up the history of this branch, accidentally synced after rebasing. Moved changes to new branch and opened a new PR #98

**Pull request closed without merging**